File: src/p0007/cpp/solution.cpp
#include <iostream>
using namespace std;


class Solution {
public:
    int reverse(int x) {
        int y = 0, n;
        for (; x; x /= 10) {
            n = y * 10 + x % 10;
            if ((n - x % 10) / 10 != y) {
                y = 0; break;
            }
            y = n;
        }
        return y;
    }
};

File: src/p0007/python/solution.py
class Solution:
    def reverse(self, x: int) -> int:
        if x < 0:
            y = -int(str(-x)[::-1])
            if y < -(2 ** 31):
                y = 0
        else:
            y = int(str(x)[::-1])
            if y >= (2 ** 31):
                y = 0
        return y

